### PR TITLE
Add switch for command line & test.sh to skip smt

### PR DESCRIPTION
**Pull request by soroosh-sdi**
- add --no-smt in scripts/tests.sh and test/cmdlineTests.sh

Signed-off-by: soroosh-sdi <soroosh.sardari@gmail.com>

### CODE REVIEW

File: test/cmdlineTests.sh [hunk 1]
```diff
@@ -302,25 +320,25 @@ test_solc_behaviour "${0}" "ctx:=/some/remapping/target" "" "" 1 "" "Invalid rem
 printTask "Running general commandline tests..."
 (
     cd "$REPO_ROOT"/test/cmdlineTests/
-    for tdir in ${*:-*/}
+    (( ${#selected_tests[@]} > 0 )) || selected_tests=(*)
```
Comment (soroosh-sdi) [inline test/cmdlineTests.sh:323]: > What's the purpose of this? If there are no selected tests, select all tests?
Shouldn't it be (( ${#selected_tests[@]} == 0 )) && selected_tests=(*) or (( ${#selected_tests[@]} > 0 )) || selected_tests=(*) for that?

sorry for that, it is fixed now.

**Comment by soroosh-sdi** [comment]
@cameel 
this is the PR we talked about in #11873.

**Review by cameel — APPROVED** [review]

**Review by leonardoalt — APPROVED** [review]

**Comment by leonardoalt** [comment]
I guess we don't have tests for this option? Should be fine as long as the normal ones still work, and we only use it locally.

**Comment by cameel** [comment]
Well, I think we don't generally test the test scripts :)

And yeah, the option is meant mostly for local use. We do not really have a use case for it in CI so far, which is why no one added it so far.

**Comment by leonardoalt** [comment]
Yea my point was more to be sure we're not breaking the test script for the general use.

**Pull request merged**